src/Microsoft.Azure.WebJobs.ServiceBus/Config/ServiceBusExtensionConfig.cs

```diff
@@ -51,23 +51,15 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            // Register an exception handler for background exceptions
-            // coming from MessageReceivers.
-            //
-            // The message options is a host level instance that is shared
-            // across all bindings, so we have to subscribe to it at the
-            // host level.
-            Config.MessageOptions.ExceptionReceived += (s, e) =>
-            {
-                Utility.LogExceptionReceivedEvent(e, "MessageReceiver", context.Trace, context.Config.LoggerFactory);
-            };
-
             // get the services we need to construct our binding providers
             INameResolver nameResolver = context.Config.GetService<INameResolver>();
             IExtensionRegistry extensions = context.Config.GetService<IExtensionRegistry>();
 
+            // register the background exception handler
+            var exceptionHandler = MessagingExceptionHandler.Subscribe(Config.MessageOptions, context.Trace, context.Config.LoggerFactory);
```

Comment:
The key for ServiceBus issue fix was to capture the exception handler and flow it down to the listener, so it can be unsubscribed BEFORE we abort the messaging factory.